eclipse-theia/theia
# Fix #4783: Support flatten/compress folders in the explorer

Signed-off-by: Esther Perelman <esther.perelman@sap.com>


#### What it does

Fix #4783

#### How to test

**A.**

 1. Go to preferences
 2. Search for `compact folders`  
 3. Toggle the setting
 4. See that folders are compressed/uncompressed accordingly...

**B.** 

- Add/Remove folders 
- Right click on compressed element
- Search any compressed folder (stand on a folder in the tree and type the letters to search)

![compressible sample — Theia Browser Example - Google Chrome 2021-10-13 14-52-08](https://user-images.githubusercontent.com/71729183/137128386-92a38cf1-afb1-4bfe-81c1-a14ff3339c28.gif)

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
